Threads.md

```diff
@@ -0,0 +1,403 @@
+# Threads
+
+This page describes a proposal for the post-MVP
+[threads feature][future threads].
+
+## Modifications to [Portability.md](Portability.md)
+
+...
+
+Only 32-bit atomic accesses are guaranteed to be lock-free. This matches the
+[ECMAScript specification](https://tc39.github.io/ecma262/#sec-atomics.islockfree).
+
+* Availability of lock-free atomic memory operators, when naturally aligned,
+  for 32-bit accesses. At a minimum this must include an atomic
+  compare-and-exchange operator (or equivalent load-linked/store-conditional).
+
+...
+
+## Modifications to [Semantics.md](Semantics.md)
+
+...
+
+If the effective address of a memory access is a multiple of the alignment
+attribute value of the memory access, the memory access is considered
+*aligned*, otherwise it is considered *misaligned*.
+
+Misaligned atomic accesses trap. For non-atomic accesses, aligned and
+misaligned accesses have the same behavior.
+
+Alignment of non-atomic accesses affects performance as follows:
+
+...
+
+## Additions to [Semantics.md](Semantics.md)
+
+### Shared Linear Memory
+
+A Linear memory can be marked as shared, which allows it to be shared between
+threads of execution. The shared memory can be imported or defined in the
+module. It is a validation error to attempt to import shared linear
+memory if the module's memory import doesn't specify that it allows shared
+memory.
```

Comment:
It'd be good to flesh this out a bit more, including [these sorts of details](https://github.com/WebAssembly/design/issues/950#issuecomment-272360540).